packages/dx-core/src/utils.js

```diff
@@ -1,13 +1,16 @@
-export const sortPlugins = (plugins) => {
-  const result = plugins.slice();
-  result.sort((a, b) => {
-    const aPosition = a.position();
-    const bPosition = b.position();
-    for (let i = 0; i < Math.min(aPosition.length, bPosition.length); i += 1) {
-      if (aPosition[i] < bPosition[i]) return -1;
-      if (aPosition[i] > bPosition[i]) return 1;
-    }
-    return aPosition.length - bPosition.length;
-  });
+const compare = (a, b) => {
+  const aPosition = a.position();
+  const bPosition = b.position();
+  for (let i = 0; i < Math.min(aPosition.length, bPosition.length); i += 1) {
+    if (aPosition[i] < bPosition[i]) return -1;
+    if (aPosition[i] > bPosition[i]) return 1;
+  }
+  return aPosition.length - bPosition.length;
+};
+
+export const insertWithSorting = (newItem, array) => {
```

Comment:
Slightly strange API. Typically we specify the array to work with in the first parameter. What do you think?